tools/timemachine/miniserver/miniServer.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2022 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package miniserver
+
+import (
+	"context"
+	"os"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/iotexproject/iotex-core/action/protocol"
+	"github.com/iotexproject/iotex-core/blockchain/block"
+	"github.com/iotexproject/iotex-core/blockchain/genesis"
+	"github.com/iotexproject/iotex-core/chainservice"
+	"github.com/iotexproject/iotex-core/config"
+	"github.com/iotexproject/iotex-core/p2p"
+	"github.com/iotexproject/iotex-core/pkg/log"
+	"github.com/iotexproject/iotex-core/state/factory"
+)
+
+// const
+const (
+	// config files
+	_configPath  = "./tools/timemachine/etc/config.yaml"
+	_genesisPath = "./tools/timemachine/etc/genesis.yaml"
+)
+
+type (
+	// MiniServer is an instance to build chain service
+	MiniServer struct {
+		cs         *chainservice.ChainService
+		stopHeight uint64
+	}
+
+	// Option sets MiniServer construction parameter
+	Option func(*MiniServer)
+)
+
+// WithStopHeightOption sets the stopHeight
+func WithStopHeightOption(stopHeight uint64) Option {
+	return func(svr *MiniServer) {
+		svr.stopHeight = stopHeight
+	}
+}
+
+// NewMiniServer creates instace and runs chainservice
+func NewMiniServer(cfg config.Config, opts ...Option) (*MiniServer, error) {
+	svr := &MiniServer{}
+	for _, opt := range opts {
+		opt(svr)
+	}
+
+	builder := chainservice.NewBuilder(cfg)
```

Comment:
IMO, `blockDAO` with its sub-components could serve this tool. No need to setup an chain for it.